sdk/src/main/java/com/mapbox/maps/renderer/FpsManager.kt

```diff
@@ -0,0 +1,180 @@
+package com.mapbox.maps.renderer
+
+import android.os.Handler
+import android.os.Looper
+import androidx.annotation.AnyThread
+import androidx.annotation.WorkerThread
+import androidx.core.os.postDelayed
+import com.mapbox.maps.BuildConfig
+import com.mapbox.maps.logI
+import com.mapbox.maps.logW
+import kotlin.math.pow
+
+internal class FpsManager {
+  private var userDefinedRefreshRate = -1
+  private var userDefinedRatio = 0.0
+
+  private var deviceRefreshRate = -1
+  private var deviceRefreshPeriodNs = -1L
+  private var previousRefreshNs = -1L
+  private var previousDrawnFrameIndex = 0
+  private var frameRenderTimeAccumulatedNs = 0L
+
+  private var preRenderTimeNs = -1L
+  private var choreographerTicks = 0
+  private var choreographerSkips = 0
+
+  @Volatile
+  private var fpsChangedListener: OnFpsChangedListener? = null
+
+  private var handler: Handler? = null
+
+  @WorkerThread
+  fun prepareHandler() {
+    Looper.myLooper()?.let { looper ->
+      handler = Handler(looper)
+    } ?: logW(
+      TAG,
+      "Looper could not be found for Mapbox render thread! " +
+        "Some FPS reports may be inaccurate."
+    )
+  }
+
+  @WorkerThread
+  fun setScreenRefreshRate(screenRefreshRate: Int) {
+    deviceRefreshPeriodNs = ONE_SECOND_NS / screenRefreshRate
```

Comment:
Can `screenRefreshRate` be 0? Maybe add a condition just to be safe ?

Reply:
good point, will do